### PR TITLE
[PRM-1469] Added a controller and endpoint for 3rd party API 1812 call

### PR DESCRIPTION
**Pull request by ruthdas03**

### CODE REVIEW

File: app/config/AppConfig.scala [hunk 1]
```diff
@@ -78,6 +78,11 @@ class AppConfig @Inject()(val config: Configuration, servicesConfig: ServicesCon
     else etmpBase + s"/penalty/financial-data/VRN/$vrn/VATC"
   }
 
+  def getPenaltyDetailsUrlv3(vrn: String): String = {
+    if(!isEnabled(CallAPI1812ETMP)) stubBase + s"/penalties-stub/penalties/penalty-details/VAT/VRN/$vrn"
+    else etmpBase + s"/penalties/penalty-details/VAT/VRN/$vrn"
+  }
+
```
Comment (LewisMitchell) [inline app/config/AppConfig.scala:85]: Could you not use the getPenaltyDetailsUrl method?

Comment (LewisMitchell) [reply app/config/AppConfig.scala:85]: Most likely can be deleted if changed in the connector

File: app/connectors/v3/getPenaltyDetails/GetPenaltyDetailsConnector.scala [hunk 2]
```diff
@@ -42,8 +42,7 @@ class GetPenaltyDetailsConnector @Inject()(httpClient: HttpClient,
 
   def getPenaltyDetailsForAPI(vrn: String, dateLimit: Option[String])(implicit hc: HeaderCarrier): Future[HttpResponse] = {
     val queryParam: String = s"${dateLimit.fold("")(dateLimit => s"?dateLimit=$dateLimit")}"
-    val url = appConfig.getPenaltyDetailsUrl + vrn + queryParam
-    httpClient.GET[HttpResponse](url, headers = headers).recover {
+    httpClient.GET[HttpResponse](appConfig.getPenaltyDetailsUrlv3(vrn) + queryParam, headers = headers).recover {
```
Comment (LewisMitchell) [inline app/connectors/v3/getPenaltyDetails/GetPenaltyDetailsConnector.scala:45]: Could you not use the getPenaltyDetailsUrl method?

File: app/controllers/APIController.scala [hunk 3]
```diff
@@ -172,6 +174,26 @@ class APIController @Inject()(etmpService: ETMPService,
             Status(res.status)(Json.toJson(res.body))
           case _ =>
             logger.warn(s"[APIController][getFinancialDetails] status ${res.status} returned from EIS " +
+                s"Status code:'${res.status}', Body: '${res.body}")
```
Comment (LewisMitchell) [inline app/controllers/APIController.scala:177]: Is this still needed? Do we want to log the body? There could be sensitive data

File: test/connectors/v3/getPenaltyDetails/GetPenaltyDetailsConnectorSpec.scala [hunk 4]
```diff
@@ -148,7 +149,7 @@ class GetPenaltyDetailsConnectorSpec extends SpecBase {
     val queryParam = "?dateLimit=09"
 
     "return a 200 when the call succeeds" in new Setup {
-      when(mockHttpClient.GET[HttpResponse](Matchers.eq(s"/penalty/details/VATC/VRN/123456789$queryParam"),
+      when(mockHttpClient.GET[HttpResponse](Matchers.eq(s"/penalties/penalty-details/VAT/VRN/123456789$queryParam"),
```
Comment (LewisMitchell) [inline test/connectors/v3/getPenaltyDetails/GetPenaltyDetailsConnectorSpec.scala:152]: I think these can be reverted back?

File: test/controllers/APIControllerSpec.scala [hunk 5]
```diff
@@ -511,4 +513,112 @@ class APIControllerSpec extends SpecBase with FeatureSwitching {
...
+          |       "LPP1LRPercentage": 2.00,
+          |       "LPP1HRPercentage": 2.00,
+          |       "penaltyChargeDueDate": "2022-10-30"
+          |   }]
+          | }
+          |}
+          |""".stripMargin)
+      when(mockGetPenaltyDetailsConnector.getPenaltyDetailsForAPI(any(), any())(any()))
+        .thenReturn(Future.successful(HttpResponse.apply(OK, sampleAPI1812Response.toString)))
+      val result = controller.getPenaltyDetails(vrn ="123456789", dateLimit = Some("02"))(fakeRequest)
```
Comment (LewisMitchell) [inline test/controllers/APIControllerSpec.scala:601]:
```suggestion
      val result = controller.getPenaltyDetails(vrn = "123456789", dateLimit = Some("02"))(fakeRequest)
```

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just some changes to make

File: app/config/AppConfig.scala (hunk 1 above)
Comment (ruthdas03) [reply app/config/AppConfig.scala:85]: > Could you not use the getPenaltyDetailsUrl method?

this method returns url '/penalty/details/VATC/VRN/' used in getPenaltyDetails method.

Comment (LewisMitchell) [reply app/config/AppConfig.scala:85]: But the connector call shouldn't be different to what we use?

File: app/controllers/APIController.scala (hunk 3 above)
Comment (ruthdas03) [reply app/controllers/APIController.scala:177]: Ok. So can we keep the status code and remove the body from the logs.

**Comment by ruthdas03** [comment]
Implemented the changes requested.

File: app/controllers/APIController.scala (hunk 3 above)
Comment (LewisMitchell) [reply app/controllers/APIController.scala:177]: I think this should still be removed, we already have the status logged on line 176.

File: app/controllers/APIController.scala [hunk 6]
```diff
@@ -172,7 +174,27 @@ class APIController @Inject()(etmpService: ETMPService,
...
+        res => res.status match {
+          case OK =>
+            logger.debug("[APIController][getPenaltyDetails] Ok response received: " + res)
+            Ok(res.json)
+          case NOT_FOUND =>
+            logger.debug("[APIController][getPenaltyDetails] Error received: " + res)
+            Status(res.status)(Json.toJson(res.body))
+          case _ =>
+            logger.warn(s"[APIController][getPenaltyDetails] status ${res.status} returned from EIS " +
+              s"Status code:'${res.status}'")
```
Comment (LewisMitchell) [inline app/controllers/APIController.scala:197]: The status is already being logged on 196

**Review by LewisMitchell — DISMISSED** [review]

**Pull request merged**